src/compiler/commandLineParser.ts

```diff
@@ -1456,6 +1468,10 @@ namespace ts {
 
     /*@internal*/
     export const compilerOptionsDidYouMeanDiagnostics: ParseCommandLineWorkerDiagnostics = {
+        alternateMode: {
+            diagnostic: Diagnostics.Compiler_option_0_may_only_be_used_with_build,

```

Comment:
Because this is not going to be normal flow, rather use `getOptionsNameMap` that reuses map instead of creating new set/map